File: leetcode/archived/C++/Valid_parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stack1;
        
        for(int i = 0; i != s.size(); i++)
        {
            if((s[i] == '(')|| (s[i] == '{') || (s[i] == '[') || (i == 0))
                stack1.push(s[i]);
            else
            {
                if(s[i] == ')')
                {
                    if((stack1.empty()) || ('(' != stack1.top()))
                        return false;
                    else
                        stack1.pop();
                }
                if(s[i] == '}')
                {
                    if((stack1.empty()) || ('{' != stack1.top()))
                        return false;
                    else
                        stack1.pop();
                }
                if(s[i] == ']')
                {
                    if((stack1.empty()) || ('[' != stack1.top()))
                        return false;
                    else
                        stack1.pop();
                }
            } 
        }
        
        if(stack1.empty())
            return true;
        else
            return false;
    }
};

File: leetcode/datastructure/stack/Valid Parentheses.go
func isValid(s string) bool {
    stk := []byte{}
    for i := 0; i < len(s); i++{
        if s[i] == '(' || s[i] == '[' ||
        s[i] == '{'{
            stk = append(stk, s[i])
        }else{
            if len(stk) == 0{
                return false
            }
            if s[i] == ')' && stk[len(stk)-1] == '('{
                stk = stk[0:len(stk)-1]
            }else if s[i] == ']' && stk[len(stk)-1] == '['{
                stk = stk[0:len(stk)-1]
            }else if s[i] == '}' && stk[len(stk)-1] == '{'{
                stk = stk[0:len(stk)-1]
            }else{
                return false
            }
        }
    }
    if len(stk) > 0{
        return false
    }
    return true
}